bin/create-digests.py

```diff
@@ -117,11 +117,388 @@ def collect_digests(digests, subdir, exclude=[]):
                         digests[filepath] = {}
                     digests[filepath] = digest.hexdigest()
 
+def add_totals(digests, total_key='total_contributions', total_group_key=None, total_subkey=None, filepath='build/_data/totals.json'):
+    ''' Sum totals from build/_data/totals.json and add to build/digests.json
+    
+    This method will look for values for the key specified by `total_key` in the
+    totals.json file, which contains totals for all elections, and add them
+    all together to get an overall total and save it to build/digests.json.  The
+    totals for each election are also saved.  All the numbers are grouped in
+    build/digests.json under the `total_group_key` key if specified.  Otherwise,
+    the numbers are groupd in build/digests.json under the `total_key` key.
+    '''
+
+    # prepare location to save totals
+    if total_subkey is None:
+        full_total_key = f'_{total_key}_from_totals'
+    else:
+        full_total_key = f'_{total_key}_{total_subkey}_from_totals'
+    if total_group_key is None:
+        total_group_key = total_key
+    if not f'_{total_group_key}' in digests:
+        digests[f'_{total_group_key}'] = {}
+    # sum totals and save to digests
+    with open(filepath, 'r', encoding='utf-8') as fp:
+        #logging.info(filepath)
+        data = json.load(fp)
+        total = 0
+        for election_name in data.keys():
+            election = data[election_name]
+            if not election_name in digests[f'_{total_group_key}']:
+                digests[f'_{total_group_key}'][election_name] = {}
+            election_info = election[total_key]
+            election_total = 0
+            if type(election_info) == dict:
+                if total_subkey is None:
+                    for key in election_info:
+                        election_total += election_info[key]
+                else:
+                    election_total += election_info.get(total_subkey,0) or 0
+            else:
+                election_total = election_info
+            digests[f'_{total_group_key}'][election_name][full_total_key] = round(election_total,2)
+            total += election_total
+        digests[f'_{total_group_key}'][full_total_key] = round(total,2)
+
+def add_tickets_total(digests, ticket_type='candidates', total_key='total_contributions', total_group_key=None, total_subkey=None):
```

Comment:
Why is this called "tickets"?  It seems the ticket_type is referring to the type of filer either candidate or referendum.  I think something like election_type would be more intuitive.